apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.YCSB/Program.cs

```diff
@@ -0,0 +1,513 @@
+                }
+            }
+
+            var timeout = Task.Delay(TimeSpan.FromSeconds(10));
+            await Task.WhenAny(Task.WhenAll(all), timeout);
+            s_totaloperations = all.Count;
+        }
+
+        /// <summary>
+        /// Runs a single worker task
+        /// </summary>
+        /// <param name="numWorkers"></param>
+        private static async Task RunWithTargetWorkers(int numWorkers)
+        {
+            //yield to let Main continue through.
+            await Task.Yield();
+            var parallelWorkers = new List<Task>();
+            var operation = ChooseOperation();
+
+            while (Interlocked.Decrement(ref s_operationsRemaining) > 0)
+            {
+                parallelWorkers.Add(s_operations[Math.Min(operation, s_operations.Count - 1)].Action(
+                    s_operations[Math.Min(operation, s_operations.Count - 1)].Latencies));
+                if (s_operationsRemaining % 1000 == 0)
+                {
+                    DebugMessage($"{s_operationsRemaining} operations remaining.");
+                }
+                operation = ChooseOperation();
+                if (parallelWorkers.Count >= numWorkers)
+                {
+                    try
+                    {
+                        parallelWorkers.Remove(await Task.WhenAny(parallelWorkers));
+                    }
+                    catch (Exception e)
+                    {
+                        DebugMessage($"This error is being ignored and we are continuing the test: {e.Message}");
+                        parallelWorkers.Clear(); //start over.
+                    }
+                }
+            }
+        }
+
+        private static int ChooseOperation()
+        {
+            var operationTarget = s_rand.Value.NextDouble() * s_total;
+            var operation = 0;
+            operationTarget -= s_operations[operation].Proportion;
+            while (operationTarget > 0 && operation < s_operations.Count)
+            {
+                operation++;
+                if (operation < s_operations.Count)
+                {
+                    operationTarget -= s_operations[operation].Proportion;
+                }
+            }
+            return operation;
+        }
+
+        private static void PrintResults(long totalElapsedMillis)
+        {
+            Console.WriteLine($"[OVERALL], RunTime(ms), {totalElapsedMillis}");
+            Console.WriteLine(
+                $"[OVERALL], Throughput(ops/sec), {1000 * s_totaloperations / totalElapsedMillis}");
+
+            foreach (var operation in s_operations)
+            {
+                var sortedlatencies = operation.Latencies.DefaultIfEmpty(0).OrderByDescending(x => x).ToList();
+                Console.WriteLine($"[{operation.Name}], Operations, {operation.Latencies.Count}");
+                Console.WriteLine(
+                    $"[{operation.Name}], AverageLatency(us), {operation.Latencies.DefaultIfEmpty(0).Average() * 1000}");
```

Comment:
Can't we just use `sortedlatencies` for all these? (Indeed, won't the MinLatency be `sortedlatencies.Last()`, and MaxLatency be `sortedlatencies.First()`?)